smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/StructureGenerator.kt

```diff
@@ -216,83 +202,35 @@ class StructureGenerator(
         // situation we have with constructors and positional arguments not playing well
         // with models evolving over time (e.g. new fields in different positions)
         writer.write("")
-            .write("fun copy(block: DslBuilder.() -> #Q = {}): #class.name:L = BuilderImpl(this).apply(block).build()", KotlinTypes.Unit)
+            .write("inline fun copy(block: Builder.() -> #Q = {}): #Q = Builder(this).apply(block).build()", KotlinTypes.Unit, symbol)
             .write("")
     }
 
-    private fun renderJavaBuilderInterface() {
+    private fun renderBuilder() {
         writer.write("")
-            .withBlock("interface FluentBuilder {", "}") {
-                write("fun build(): #class.name:L")
+            .withBlock("class Builder {", "}") {
                 for (member in sortedMembers) {
                     val (memberName, memberSymbol) = memberNameSymbolIndex[member]!!
                     // we want the type names sans nullability (?) for arguments
                     writer.renderMemberDocumentation(model, member)
                     writer.renderAnnotations(member)
-                    write("fun #1L(#1L: #2T): FluentBuilder", memberName, memberSymbol)
-                }
-            }
-    }
-
-    private fun renderDslBuilderInterface() {
-        writer.write("")
-            .withBlock("interface DslBuilder {", "}") {
-                val structMembers: MutableList<MemberShape> = mutableListOf()
-
-                for (member in sortedMembers) {
-                    val (memberName, memberSymbol) = memberNameSymbolIndex[member]!!
-                    val targetShape = model.getShape(member.target).get()
-                    when {
-                        targetShape.isStructureShape -> structMembers.add(member)
-                    }
-
-                    writer.renderMemberDocumentation(model, member)
-                    writer.renderAnnotations(member)
-                    write("var #L: #P", memberName, memberSymbol)
+                    write("var #L: #E", memberName, memberSymbol)
```

Comment:
fix

this is wrong. if you look at the previous code we sorted out structure members and then generate a DSL overload for them in addition to generating the property. We did this on the `DslBuilder` inteface with a default implementation so the `BuilderImpl` didn't actually have to override anything.

See test [here](https://github.com/awslabs/smithy-kotlin/pull/534/files#diff-f65136e1adc3c3118534445fe571fad8089d990c4712041c9b0b6a4aab9938a7L206) where it has been removed.

Example:
```
structure Baz { ... }
structure Foo {
     intMember: Integer
     structMember: Baz
}
```

Should generate something like:

```kotlin
class Foo {
      ...


      class Builder {
           var intMember: Integer? = null
           var structMember: Baz? = null

            // this is now gone and needs fixed
            fun baz(block: Baz.Builder.() -> kotlin.Unit): Unit { this.structMember = Baz.invoke(block) }

      }
}
```

Reply:
Yep, good catch.  I looked at adding a compile test to fix this but the original issue is that I mistakenly removed the expected output from the codegen test.  So the issue was caught because you saw the diff was incorrect due to the bad test update.  As such I left the compile tests as-is.